src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -977,24 +966,30 @@ public Map<String, TableSnapshot> listSnapshots()
             String tag = entry.getKey();
             Set<File> snapshotDirs = entry.getValue();
             SnapshotManifest manifest = maybeLoadManifest(metadata.keyspace, metadata.name, tag, snapshotDirs);
-            snapshots.put(tag, buildSnapshot(tag, manifest, snapshotDirs));
+            boolean ephemeral = manifest == null ? isEphemeral(snapshotDirs) : manifest.ephemeral;
+            snapshots.put(tag, buildSnapshot(tag, manifest, snapshotDirs, ephemeral));
         }
 
         return snapshots;
     }
 
-    protected TableSnapshot buildSnapshot(String tag, SnapshotManifest manifest, Set<File> snapshotDirs) {
+    protected TableSnapshot buildSnapshot(String tag, SnapshotManifest manifest, Set<File> snapshotDirs, boolean ephemeral) {
         Instant createdAt = manifest == null ? null : manifest.createdAt;
         Instant expiresAt = manifest == null ? null : manifest.expiresAt;
         return new TableSnapshot(metadata.keyspace, metadata.name, metadata.id.asUUID(), tag, createdAt, expiresAt,
-                                 snapshotDirs);
+                                 snapshotDirs, ephemeral);
+    }
+
+    protected static boolean isEphemeral(Set<File> snapshotDirs)
```

Comment:
Can you add a short comment about why this method is needed - should we remove it at some point and just use the manifest to decide if a snapshot is ephemeral?

Reply:
ok i ll add comment

Reply:
I rewrote it so that method is needed anymore.